bundles/org.openhab.binding.pjlinkdevice/src/main/java/org/openhab/binding/pjlinkdevice/internal/PJLinkDeviceHandler.java

```diff
@@ -0,0 +1,329 @@
+import org.openhab.binding.pjlinkdevice.internal.device.command.power.PowerQueryResponse.PowerQueryResponseValue;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+
+/**
+ * The {@link PJLinkDeviceHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Nils Schnabel - Initial contribution
+ */
+@NonNullByDefault
+public class PJLinkDeviceHandler extends BaseThingHandler {
+
+  private @Nullable PJLinkDeviceConfiguration config;
+
+  private @Nullable PJLinkDevice device;
+
+  private InputChannelStateDescriptionProvider stateDescriptionProvider;
+
+  private final Logger logger = LoggerFactory.getLogger(PJLinkDeviceHandler.class);
+
+  private @Nullable ScheduledFuture<?> refreshJob;
+
+  private @Nullable ScheduledFuture<?> setupJob;
+
+  public PJLinkDeviceHandler(Thing thing, InputChannelStateDescriptionProvider stateDescriptionProvider) {
+    super(thing);
+    this.stateDescriptionProvider = stateDescriptionProvider;
+  }
+
+  @Override
+  public void dispose() {
+    clearSetupJob();
+
+    clearRefreshInterval();
+
+    PJLinkDevice device = this.device;
+    if (device != null) {
+      device.dispose();
+    }
+
+    this.config = null;
+    this.device = null;
+  }
+
+  public void refresh(PJLinkDeviceConfiguration config) {
+    // Do not poll if configuration is incomplete
+    if (PJLinkDeviceHandler.this.getThing().getStatusInfo()
+        .getStatusDetail() != ThingStatusDetail.CONFIGURATION_ERROR) {
+
+      PJLinkDeviceHandler.this.logger.debug("Polling device status...");
+      if (config.refreshPower) {
+        PJLinkDeviceHandler.this.handleCommand(new ChannelUID(getThing().getUID(), CHANNEL_POWER), RefreshType.REFRESH);
+      }
+      if (config.refreshMute) {
+        // this updates both CHANNEL_AUDIO_MUTE and CHANNEL_VIDEO_MUTE
+        PJLinkDeviceHandler.this.handleCommand(new ChannelUID(getThing().getUID(), CHANNEL_AUDIO_MUTE),
+            RefreshType.REFRESH);
+      }
+      if (config.refreshInputChannel) {
+        PJLinkDeviceHandler.this.handleCommand(new ChannelUID(getThing().getUID(), CHANNEL_INPUT), RefreshType.REFRESH);
+      }
+    }
+  }
+
+  public PJLinkDevice getDevice() throws UnknownHostException, ConfigurationException {
+    PJLinkDevice device = this.device;
+    if (device == null) {
+      PJLinkDeviceConfiguration config = getConfiguration();
+      this.device = device = new PJLinkDevice(config.tcpPort, InetAddress.getByName(config.ipAddress),
+          config.adminPassword);
+    }
+    return device;
+  }
+
+  @Override
+  public void handleCommand(ChannelUID channelUID, Command command) {
+    logger.trace("Received command {} on channel {}", command, channelUID.getId());
+    try {
+      PJLinkDevice device = getDevice();
+      switch (channelUID.getId()) {
+        case CHANNEL_POWER:
+          logger.trace("Received power command" + command);
```

Comment:
```suggestion
          logger.trace("Received power command {}", command);
```
Don't use string concatenation in loggers but placeholders. Also applies to other logging statements.